### PR TITLE
cmake: FreeBSD specific excludes in CMakeLists.txt part 2

### PR DESCRIPTION
**Pull request by wjwithagen**
Changes to the CMakeLists.txt files specific for FreeBSD building.
These are the items that can not (yet) be build on FreeBSD
Excluding build target based on
-   LINUX
-   RBD
-   AIO
-   LIBCEPHFS
-   KEYUTILS_FOUND

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -696,9 +696,11 @@ install(PROGRAMS
 
 add_subdirectory(bash_completion)
 
-set(parse_secret_files
-  common/secret.c)
-add_library(parse_secret_objs OBJECT ${parse_secret_files})
+if(LINUX)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: @wjwithagen could you please note why this is not `secret.c` is not buildable in FreeBSD? for example, keyutils is not available in FreeBSD?

and why krbd is not applicable to FreeBSD?

File: src/test/objectstore/CMakeLists.txt [hunk 2]
```diff
@@ -88,12 +88,14 @@ target_link_libraries(ceph_test_filestore_idempotent_sequence
...
   DESTINATION ${CMAKE_INSTALL_BINDIR})
 
-# unittest_chain_xattr
-add_executable(unittest_chain_xattr
-  chain_xattr.cc
-  )
-add_ceph_unittest(unittest_chain_xattr ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/unittest_chain_xattr)
-target_link_libraries(unittest_chain_xattr os global)
+if(WITH_AIO)
+  # unittest_chain_xattr
```
Comment (tchaikov) [inline src/test/objectstore/CMakeLists.txt]: i don't really understand how chain_xattr is related to AIO?

**Comment by wjwithagen** [comment]
@tchaikov 
You are correct.
compilation yesterday generated `bit_alloc` missing errors, which depends on Bluestore.
But I misread the multithreaded build logging
Should be fixed now.

File: src/CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [inline src/CMakeLists.txt]: @wjwthagen this is not addressed.

**Comment by wjwithagen** [comment]
@tchaikov 
I missed that first remark. Sorry about that.
Reworded the commit for src/CMakeLists.txt

File: src/CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [inline src/CMakeLists.txt]: i guess maybe we can check for `KEYUTILS_FOUND` instead?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -696,7 +696,7 @@ install(PROGRAMS
 
 add_subdirectory(bash_completion)
 
-if(LINUX)
+if(KEYUTILS_FOUND)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: could you squash this commit into 05d72fb45eb8ebf9c5932dab88fa0815b49c4e2f ?

Comment (wjwithagen) [inline src/CMakeLists.txt]: @tchaikov 
Done

**Comment by tchaikov** [comment]
pushed to gb as wip-10973. once gb are happy, it's good to merge.

**Comment by wjwithagen** [comment]
@tchaikov 
Thanx

**Comment by wjwithagen** [comment]
jenkins, retest this please

**Pull request merged**